### PR TITLE
docs: updated fork process

### PR DESCRIPTION
**Pull request by Eliastisys**
> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [x] personal data beyond what is necessary for interacting with this pull request, nor
> - [x] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [ ] kind/improvement   
- [ ] kind/deprecation   
- [x] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change 
- [ ] kind/dev-change   
- [ ] kind/security     
- [ ] [kind/adr]()      

### What does this PR do / why do we need this PR?

Updated process of handling the kubespray fork.
Create one release branch `release-X.Y.Z-ck8s` in our fork per upstream release `X.Y.Z`. We then tag the latest commit of that release branch with `vX.Y.Z-ck8s<patch-number>`.

#### Checklist

- [ ] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [ ] The change requires migration steps
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [ ] The logs do not show any errors after the change
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [ ] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Network Policy checks:
  - [ ] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packets in the `NetworkPolicy Dashboard`
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

**Review by robinAwallace — APPROVED** [review]

File: docs/managing-fork.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+# Managing kubespray fork
```
Comment (OlleLarsson) [inline docs/managing-fork.md:1]: Nit: shall we name this document `managing-kubespray-fork.md`?

File: docs/managing-fork.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+Make sure the master branch is up to date with the latest upstream release version:
+
+```console
+$ cd kubespray
+$ git fetch --all
+$ git switch master
+
+$ git status # Ensure these lines match
+On branch master
+Your branch is up to date with 'origin/master'.
```
Comment (OlleLarsson) [inline docs/managing-fork.md:65]: So if the output match I should stop and proceed to the step 3?

Comment (Eliastisys) [reply docs/managing-fork.md:65]: yes, should i make it more clear?

Comment (OlleLarsson) [reply docs/managing-fork.md:65]: I would say yes if possible :)

File: docs/managing-fork.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+Tag the latest commit in the fork release branch. `patch-number` starts at 1 and increments by one for every new `compliantkubernetes-kubespray` release.
+
+```sh
+cd <fork-directory>
+git switch "release-X.Y.Z-ck8s"
+git pull
+git tag -a "vX.Y.Z-ck8s<patch-number>"
+git push --tags
+```
```
Comment (aarnq) [inline docs/managing-fork.md:152]: Isn't it better if we do this during the release step?

Comment (Eliastisys) [reply docs/managing-fork.md]: Yes, it's included in the release steps, I just kept it here to sort of give the whole picture. Added a disclaimer now, or should I remove it from here entirely?

File: docs/managing-kubespray-fork.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
+# Managing kubespray fork
+
+This document details the process to upgrade and/or make changes to the kubespray fork.
+
+TLDR:
```
Comment (OlleLarsson) [inline docs/managing-kubespray-fork.md:5]: Nit
```suggestion
TL;DR:
```

File: docs/managing-kubespray-fork.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+The releases will follow the version of the upstream version, with the version string `<version>-ck8s`.
+
+- `<version>` - denotes the current version of the upstream version.
+- `ck8s` - added to prevent confusion with upstream release branch
+
+```sh
+git fetch --all
+
+# if a release/patch exists and you only want to add a fix to the fork:
+git switch "release-X.Y.Z-ck8s"
```
Comment (OlleLarsson) [inline docs/managing-kubespray-fork.md:104]: Nit: what do you think for consistency to use `<>` format in all snippets?
```suggestion
git switch "release-<X.Y.Z>-ck8s"
```
or make surround the entire argument in `<>`

File: docs/managing-kubespray-fork.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+git push -u origin "release-<X.Y.Z>-ck8s"
+```
+
+### Step 5
+
+Backport the fixes to the release branch.
+
+```sh
+git cherry-pick COMMIT-SHA [COMMIT-SHA ...]
```
Comment (OlleLarsson) [inline docs/managing-kubespray-fork.md:118]:
```suggestion
git cherry-pick <COMMIT-SHA> <[COMMIT-SHA ...]>
```

**Review by OlleLarsson — APPROVED** [review]
LGTM, haven't tested any of the steps this time around.
